src/hotspot/cpu/x86/stubGenerator_x86_64_poly.cpp

```diff
@@ -0,0 +1,1036 @@
+  0x00000fffffffffff, 0x00000fffffffffff,
+  0x00000fffffffffff, 0x00000fffffffffff,
+  0x00000fffffffffff, 0x00000fffffffffff,
+};
+static address poly1305_mask44() {
+  return (address)POLY1305_MASK44;
+}
+
+// Compute product for 8 16-byte message blocks,
+// i.e. For each block, compute [a2 a1 a0] = [a2 a1 a0] × [r2 r1 r0]
+//
+// Each block/number is represented by 3 44-bit limb digits, start with multiplication
+//
+//      a2       a1       a0
+// ×    r2       r1       r0
+// ----------------------------------
+//     a2×r0    a1×r0    a0×r0
+// +   a1×r1    a0×r1  5×a2×r1'     (r1' = r1<<2)
+// +   a0×r2  5×a2×r2' 5×a1×r2'     (r2' = r2<<2)
+// ----------------------------------
+//        p2       p1       p0
+//
+// Then, propagate the carry (bits after bit 44) from lower limbs into higher limbs.
+// Then, modular reduction from upper limb wrapped to lower limbs
+//
+// Math Note 1: 'carry propagation' from p2 to p0 involves multiplication by 5 (i.e. slightly modified modular reduction from above):
+//    ( p2×2^88 ) mod 2^130-5
+//                             = ( p2'×2^88 + p2''×2^130) mod 2^130-5 // Split on 130-bit boudary
+//                             = ( p2'×2^88 + p2''×2^130 - 5×p2'' + 5×p2'') mod 2^130-5
+//                             = ( p2'×2^88 + p2''×(2^130 - 5) + 5×p2'') mod 2^130-5 // i.e. adding multiples of modulus is a noop
+//                             = ( p2'×2^88 + 5×p2'') mod 2^130-5
+//
+// Math Note 2: R1P = 4*5*R1 and R2P = 4*5*R2; This precomputation allows simultaneous reduction and multiplication.
+// This is not the standard 'multiply-upper-by-5', here is why the factor is 4*5 instead of 5.
+// For example, partial product (a2×r2):
+//    (a2×2^88)×(r2×2^88) mod 2^130-5
+//                                    = (a2×r2 × 2^176) mod 2^130-5
+//                                    = (a2×r2 × 2^46×2^130) mod 2^130-5
+//                                    = (a2×r2×2^46 × 2^130- 5×a2×r2×2^46 + 5×a2×r2×2^46) mod 2^130-5
+//                                    = (a2×r2×2^46 × (2^130- 5) + 5×a2×r2×2^46) mod 2^130-5 // i.e. adding multiples of modulus is a noop
+//                                    = (5×a2×r2×2^46) mod 2^130-5
+//                                    = (a2×5×r2×2^2 × 2^44) mod 2^130-5 // Align to limb boudary
+//                                    = (a2×[5×r2×4] × 2^44) mod 2^130-5
+//                                    = (a2×R2P × 2^44) mod 2^130-5 // i.e. R2P = 4*5*R2
+//
+void StubGenerator::poly1305_multiply8_avx512(
+  const XMMRegister A0, const XMMRegister A1, const XMMRegister A2,
+  const XMMRegister R0, const XMMRegister R1, const XMMRegister R2, const XMMRegister R1P, const XMMRegister R2P)
+{
+  const XMMRegister P0_L = xmm0;
+  const XMMRegister P0_H = xmm1;
+  const XMMRegister P1_L = xmm2;
+  const XMMRegister P1_H = xmm3;
+  const XMMRegister P2_L = xmm4;
+  const XMMRegister P2_H = xmm5;
+  const XMMRegister TMP1 = xmm6;
+  const Register polyCP = r13;
```

Comment:
Could be renamed to `rscratch` (or `tmp`) since it doesn't hold constant base address anymore.

Reply:
done